### PR TITLE
openstack: remove default egress security rules

### PR DESCRIPTION
**Pull request by nhicher**
With this change, it's possible for a user to remove
the default egress security rules during the
security group creation.

It's not possible to match these rules using
os_security_group_rule module without adding a specific
condition in _find_matching_rule function (with None
values for remote_ip_prefix, protocol and ethertype).

This solution looks easier to maintain.

##### ISSUE TYPE

 - Feature Pull Request
 - New Module Pull Request
 - Bugfix Pull Request
 - Docs Pull Request

##### ANSIBLE VERSION

```

```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: lib/ansible/modules/cloud/openstack/os_security_group.py [hunk 1]
```diff
@@ -126,6 +135,9 @@ def main():
         if state == 'present':
             if not secgroup:
                 secgroup = cloud.create_security_group(name, description)
+                if egress_default_rules == 'absent':
```
Comment (emonty) [inline lib/ansible/modules/cloud/openstack/os_security_group.py]: Maybe I'm missing something, but it looks to me like setting egress_default_rules to absent will remove _all_ rules from the group that was just created (the default set of rules could be different things on different clouds iirc) Perhaps we should just make this a bool called "delete_default_rules" or something - and mention in the doc string that it only has any effect when you are creating a security group.

**Review by emonty — CHANGES_REQUESTED** [review]
Seems like a fine approach to the problem - the name of the parameter confused me at first though.

File: lib/ansible/modules/cloud/openstack/os_security_group.py (hunk 1 above)
Comment (nhicher) [reply lib/ansible/modules/cloud/openstack/os_security_group.py]: you're right, I didn't think about others default rules. I fixe the patch. Thanks for the review

**Comment by nhicher** [comment]
@emonty I did the change, but used push -f, not sure it was the right thing to do.

**Pull request closed without merging**